spyder/plugins/editor/widgets/codeeditor.py

```diff
@@ -1231,6 +1231,17 @@ def handle_go_to_definition(self, position):
             self.log_lsp_handle_errors(
                 "Error when processing go to definition")
 
+    # ------------- LSP: Code folding ranges -------------------------------
+    @request(method=LSPRequestTypes.DOCUMENT_FOLDING_RANGE)
+    def request_folding(self):
+        params = {'file': self.filename}
+        return params
+
+    @handles(LSPRequestTypes.DOCUMENT_FOLDING_RANGE)
+    def handle_folding_range(self, response):
+        ranges = response['params']
+        logger.debug(f'Ranges: {ranges}')

```

Comment:
f-strings are only available in Python 3.6+. Please use a formatting method available in Python 2.7 and 3.5+.